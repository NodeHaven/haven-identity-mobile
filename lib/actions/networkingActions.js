// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  ANALYZE_ADDRESS,
  MARK_OFFLINE,
  MARK_ONLINE,
  SAVE_GAS_PRICE
} from 'uPortMobile/lib/constants/NetworkingActionTypes'

export function goOffline () {
  return {
    type: MARK_OFFLINE
  }
}

export function goOnline () {
  return {
    type: MARK_ONLINE
  }
}

export function saveGasPrice (gasPrice) {
  return {
    type: SAVE_GAS_PRICE,
    gasPrice
  }
}

export function analyzeAddress (address) {
  return {
    type: ANALYZE_ADDRESS,
    address
  }
}
