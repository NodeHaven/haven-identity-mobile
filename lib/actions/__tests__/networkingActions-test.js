// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {goOffline, goOnline, saveGasPrice, analyzeAddress} from '../networkingActions.js'

it('creates a ANALYZE_ADDRESS action', () => {
  expect(analyzeAddress('0x1231')).toMatchSnapshot()
})

it('creates a SAVE_GAS_PRICE action', () => {
  expect(saveGasPrice(123123123123)).toMatchSnapshot()
})

it('creates a MARK_ONLINE action', () => {
  expect(goOnline()).toMatchSnapshot()
})

it('creates a MARK_OFFLINE action', () => {
  expect(goOffline()).toMatchSnapshot()
})
