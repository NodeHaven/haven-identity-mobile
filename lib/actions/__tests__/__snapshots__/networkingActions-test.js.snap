// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a ANALYZE_ADDRESS action 1`] = `
Object {
  "address": "0x1231",
  "type": "ANALYZE_ADDRESS",
}
`;

exports[`creates a MARK_OFFLINE action 1`] = `
Object {
  "type": "MARK_OFFLINE",
}
`;

exports[`creates a MARK_ONLINE action 1`] = `
Object {
  "type": "MARK_ONLINE",
}
`;

exports[`creates a SAVE_GAS_PRICE action 1`] = `
Object {
  "gasPrice": 123123123123,
  "type": "SAVE_GAS_PRICE",
}
`;
